woocommerce/woocommerce-ios
# Shipping Labels: Complete customs form UI

Part of #4687 

# Description
This PR wraps up the UI implementation of the customs form in Shipping Labels flow. Validation and submission of the form will be handled in the next PR.

# Changes
 1. Updates to reusable SwiftUI views:
   - New view `LearnMoreRow`: accepts localized text containing hyperlink and returns attributed string with the link and presents a safari webview for the link.
   - Updated `TitleAndTextFieldRow` with an initializer and default values for `keyboardType` and `symbol`.
   - Updated `SelectionList` with `selectionKeyPath` to define key path to the property to be used for selection. I wanted to use this for the country list but then used `\.self` instead - but I'm keeping this property just in case it's necessary in the future.
   - Updated `CollapsibleView` with a binding for `isCollapsed` to make the view collapsed initially when needed.
2. Updates to customs form:
- Added rows for restriction type, ATN and learn more about ATN.
- Show text fields for content type and restriction type if these are of type other.
- Added section header for package content.
- Added a list of collapsible views for items in the package. These views are collapsed by default.
3. Added view and view model for item details:
- Completed UI for item details with product description, HS Tariff number, weight, value, origin country. Values for these fields will be filled in the next PR.
- For the selection list of origin country, the country list is passed all the way from the Shipping Labels form view model through the customs form list to the item detail form. This doesn't look so neat, but at least we're saved from fetching the list again.

# Demo
![customs-form](https://user-images.githubusercontent.com/5533851/129142655-aa6ebf6f-c3d3-47f3-9093-e261344f7170.gif)

# Testing steps
1. Make sure that your test store is eligible for shipping labels (has WCShip plugin installed).
2. Create an order with international shipping (origin country different from destination country).
3. Navigate to Orders tab, select the new order and select Create Shipping Label.
4. Skip through Ship From, Ship To, Package Details to configure Customs.
5. Notice that the customs form shows default values: content type Merchandise, restriction type None, empty ATN.
6. Notice that content details text field is only shown when selecting content type Other.
7. Notice that restriction details text field is only shown when selecting restriction type Other.
8. Tapping on "Learn more" should show a web view about ATN.
9. Notice the number of item detail rows should match the number of items in the package. These rows are collapsed by default.
10. Notice that the list of countries are correct when selecting Origin Country row. Selected country should be displayed on the row when tapping Done on the list.

Update release notes:
